### ISSUE TITLE
Is there an aarch64 version of EnergyPlus? Or how to compile the aarch64 version from the source code

### ISSUE THREAD

**Issue by fubingting** [NREL/EnergyPlus#8669]

**Comment by bonnema** [comment]
Maybe start here? https://cmake.org/cmake/help/latest/manual/cmake-toolchains.7.html

**Comment by eringold** [comment]
also see: https://github.com/NREL/EnergyPlus/pull/8664

**Comment by jmarrec** [comment]
Either checkout #8664 or do this: specificy `CMAKE_OSX_ARCHITECTURES`

```
cmake -DCMAKE_OSX_ARCHITECTURES=arm64 [...]
```

Alternatively, I think this should also prefill the CMAKE_OSX_ARCHITECTURES correctly

```
export ARCHFLAGS="-arch x86_64"
cmake
```


Once the arch is detected correctly, I can assure you that EnergyPlus builds and runs fine.

**Comment by jmarrec** [comment]
@fubingting I uploaded an aarch64 version on the release page: https://github.com/NREL/EnergyPlus/releases/tag/v9.5.0

EnergyPlus-9.5.0-de239b2e5f-Darwin-macOS11.2-arm64.dmg
EnergyPlus-9.5.0-de239b2e5f-Darwin-macOS11.2-arm64.tar.gz

Could you give it a shot?

**Comment by fubingting** [comment]
> @fubingting I uploaded an aarch64 version on the release page: https://github.com/NREL/EnergyPlus/releases/tag/v9.5.0
> 
> EnergyPlus-9.5.0-de239b2e5f-Darwin-macOS11.2-arm64.dmg
> EnergyPlus-9.5.0-de239b2e5f-Darwin-macOS11.2-arm64.tar.gz
> 
> Could you give it a shot?

Thank you very much. I'll try to install it.

**Comment by fubingting** [comment]
@jmarrec

**Comment by fubingting** [comment]
@jmarrec

**Comment by fubingting** [comment]
After downloading and decompressing, I execute the three energyplus-9.5.0,EPMacro,ExpandObjects commands, and the following error occurs：

**Comment by fubingting** [comment]
After downloading and decompressing, I execute the three energyplus-9.5.0,EPMacro,ExpandObjects commands, and the following error occurs:
```
**./energyplus-9.5.0**
-bash: ./energyplus-9.5.0: cannot execute binary file: Exec format error
**file energyplus-9.5.0**
energyplus-9.5.0: Mach-O 64-bit arm64 executable, flags:<NOUNDEFS|DYLDLINK|TWOLEVEL|WEAK_DEFINES|BINDS_TO_WEAK|PIE>
```
@jmarrec

**Comment by jmarrec** [comment]
@fubingting what OS/Version are you using? This package is specifically for an M1 mac.

**Comment by fubingting** [comment]
platform：aarch64

**Comment by fubingting** [comment]
platform：aarch64
os：centos8
Because you mentioned above
```
 I uploaded an aarch64 version on the release page: https://github.com/NREL/EnergyPlus/releases/tag/v9.5.0,Could you give it a shot?
```
> @fubingting what OS/Version are you using? This package is specifically for an M1 mac.

**Comment by jmarrec** [comment]
I assumed you were using an M1 mac incorrectly. The name of the installer mentions darwin-macos11.2-arm64 so I thought it was explicit that this would work on macOS only.

We do not have a centos aarch64 package no.

**Comment by jmarrec** [comment]
#8664 was merged, so you should try to compile it from source. https://github.com/NREL/EnergyPlus/wiki/Building-EnergyPlus

You'll need cmake and a C/CXX compiler, and that should pretty much be it as far as dependencies. I suggest a fortran compiler too so you can build the auxiliary programs such as ReadVarEso and ExpandObjects.

**Comment by airallergy** [comment]
Hi @jmarrec, I tried to compile EP on M1, everything seems to work fine with binaries directly from the build folder. But after I installed the compiled package in the Applications folder, a few binaries do not work and give the error

```zsh
zsh: killed     ReadVarsESO
zsh: killed     ExpandObjects
...
```

while `bash` gives

```bash
Killed: 9
```

I did not test all binaries exhaustively. Two I found working are `energyplus` and `EPMacro`. I tried compiling 9.6, also installing the precompiled 9.5 you attached above. Any ideas why this might be?

I can open a new issue if more appropriate.
